src/features/eth.js

```diff
@@ -31,23 +32,32 @@ const Eth = () => {
   const methods = useForm({ mode: "onChange" });
   const wallet = useWallet();
   const [txDone, setTxDone] = useState(false);
+  const [openTxDetailsDialog, setOpenTxDetailsDialog] = useState(true);
+  const [hashes, setHashes] = useState([]);
   const [token1, setToken1] = useState({
     icon: "https://mindswap.finance/tokens/iq.png",
     name: "pIQ",
     precision: 3,
     chain: "Ethereum"
   });
 
+  const appendNewTxHash = txHash => setHashes(prev => prev.concat(txHash));
+
   const onSubmit = async data => {
     if (!wallet.account) {
       return;
     }
 
-    await convertPTokensTx(data.FromAmount, wallet);
+    await convertPTokensTx(data.FromAmount, wallet, appendNewTxHash);
```

Comment:
we could return txs or txs arrays as default in this kind of call to blockchain, that way we dont need to send the callback as param